### ISSUE TITLE
vinyl: on update do not read unique sk, if update mask does not match index

### ISSUE THREAD

**Issue by Gerold103** [tarantool/tarantool#2980]
On update operation at first the primary index is read. An old tuple is updated and a column mask is built. Then for unique secondary indexes are performed READ+DELETE+REPLACE. READ is used to determine, if a new tuple does not violates unique constraint of the index. But if a column mask does not match the secondary index's mask, it means, that its key is not updated, and we can avoid READ. If a key is not changed, then the uniqueness can not be broken.
For example:
```Lua
s = box.schema.create_space('test', {engine = 'vinyl'})
pk = s:create_index('pk')
sk = s:create_index('sk', {parts = {{2, 'unsigned'}}})
s:replace{1,2,3,4}
s:update({1}, {{4, '=', 100}}) -- No neccessarity to read sk. On pk update it is clear, that sk is not changed.
```

**Comment by Gerold103** [comment]
The same optimization can be applied to REPLACE. But it requires additional column mask calculation - now a REPLACE does not calculate column mask.

**Comment by locker** [comment]
Fixed by ab7260317a8c1cf0e475c091ea955c15be701ddc
